system/security/shared/caching.hpp

```diff
@@ -140,22 +140,24 @@ class CachedUser
 class CPermissionsCache : public CInterface
 {
 public:
-    CPermissionsCache(const char * _secMgrClass = nullptr)
+    CPermissionsCache(const char * _secMgrClass, ISecManager *secMgr)
     {
         m_cacheTimeoutInSeconds = DEFAULT_RESOURCE_CACHE_TIMEOUT_MINUTES * 60 * 1000;//default every hour
         m_transactionalEnabled = false;
-        m_secMgr = NULL;
+        m_secMgr = secMgr;
         m_defaultPermission = SecAccess_Unknown;
         m_secMgrClass.set(_secMgrClass);
         m_transactionalCacheTimeout = DEFAULT_CACHE_TIMEOUT_SECONDS;
     }
 
+    CPermissionsCache(ISecManager *secMgr) : CPermissionsCache(nullptr, secMgr) {}
+
     virtual ~CPermissionsCache();
 
     //Returns an owned reference to a shared cache of a given Sec Mgr class type.
     //Call this method with a unique class string ("LDAP", "MyOtherSecMgr")
     //to create a cache shared amongst security managers of the same class
-    static CPermissionsCache* getInstance(const char * _secMgrClass);
+    static CPermissionsCache* getInstance(const char * _secMgrClass, ISecManager *secMgr = nullptr);
```

Comment:
If secMgr was ever null, I think the code would now crash, but it is never null. Should remove default, and should probably pass in a reference to be clear mandatory.